File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {

        int rvsX = 0;
        int originX = x;
        
        while (x > 0) {
            rvsX *= 10;
            int num = x%10;
            rvsX += num;
            x /= 10;
        }

        if (originX < 0 || originX != rvsX) {
            return false;
        }
        
        return true;
        
    }
}

File: 0009-palindrome-number/0009-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0 : return False

        originX = x;
        reversedX = 0;

        while x > 0:
            reversedX *= 10
            num = x % 10
            reversedX += num
            x //= 10
        
        return  reversedX == originX